### ISSUE TITLE
CSS transition does not work with FA svg icons

### ISSUE THREAD

**Issue by kit246** [FortAwesome/Font-Awesome#12625]
When using Font awesome Icon with JQuery to apply class .fa-rotate-90 on click the transition property gets ignored. I assume it happens due to the SVG element being rewritten when new class is applied to the SVG element. The same thing happens when I make a copy of the class .fa-rotate-90 in a my own CSS file with the same transform property and apply this one instead. The icon is rotated correctly but without the transition happening. 
Transition property is added using custom class .menuIcon
```javascript
$(".toggleClick").click(function () {
	$(this).find(".menuIcon").toggleClass("rotate-90");
});
```
```html
<i class="fas fa-angle-right fa-fw menuIcon"></i>
```
I overcame this issue by applying a wrapper SPAN:
```html
<span class="menuIcon"><i class="fas fa-angle-right fa-fw"></i></span>
```

**Comment by tagliala** [comment]
Hi!

Thanks for being part of the Font Awesome Community.

> I assume it happens due to the SVG element being rewritten when new class is applied to the SVG element.

Yes, please go for idempotent events: https://jsfiddle.net/tagliala/qu83xrwp/25/

This request is a duplicate of #, please add a 👍  reaction to that request.

Hope it helps

Closing here

**Comment by kit246** [comment]
Hi,

Could you please explain a little more how to use the idempotent events properly?
Your example in jsfiddle doesn't help me much.
Could you please check out the difference between the 2 examples below? In the first one the span is transformed 90deg with the transition 200ms. In the second example where transition is applied directly to the SVG the element is transformed but without transition. 

SPAN Wrapper
https://jsfiddle.net/qu83xrwp/35/
No wrapper:
https://jsfiddle.net/qu83xrwp/37/

Thanks.

**Comment by tagliala** [comment]
@kit246 Sorry!

The issue is about transition and not transform 🤦‍♂️ 

Please keep using a wrapper for the moment.

There is an issue with you use case. Don't know if this could be solved or if it should be documented

@robmadole something is happening when the class list includes `fa fa-iconname`

Fails with `fa fa-home`: https://jsfiddle.net/tagliala/qu83xrwp/56/
Ok: https://jsfiddle.net/tagliala/qu83xrwp/57

**Comment by robmadole** [comment]
Yep, I think I know what's going on with this. It can probably be solved. I'll add it to the list.

**Comment by xcellsoft** [comment]
FYI, i just ran into this issue as well, what i did was reference the svg tag in the class, the transition won't on the <i> tag, it needs to affect the child <svg> tag.

`<i class="fa-thumbtack oc-thumbtack-up">`

`.oc-thumbtack-up svg {
    animation: oc-thumbtack-up-kf 200ms ease-out;
    animation-fill-mode: forwards;
    transform-origin: center;
}

@keyframes oc-thumbtack-up-kf {
    0% {
        transform: rotate(0deg);
    }
    100% {
        transform: rotate(90deg);
    }
}

.oc-thumbtack-down svg {
    animation: oc-thumbtack-down-kf 200ms ease-in;
    animation-fill-mode: forwards;
    transform-origin: center;
}

@keyframes oc-thumbtack-down-kf {
    0% {
        transform: rotate(90deg);
    }
    100% {
        transform: rotate(0deg);
    }
}
`

**Comment by flora-le** [comment]
> @kit246 Sorry!
> 
> The issue is about transition and not transform man_facepalming
> 
> Please keep using a wrapper for the moment.
> 
> There is an issue with you use case. Don't know if this could be solved or if it should be documented
> 
> @robmadole something is happening when the class list includes `fa fa-iconname`
> 
> Fails with `fa fa-home`: https://jsfiddle.net/tagliala/qu83xrwp/57/ Ok: https://jsfiddle.net/tagliala/qu83xrwp/56

The problem still exists til this day. I **can** transform the svg with **rotate**() but there is **no transition working**. 

`<svg class="svg-inline--fa fa-angle-right dropdown" data-v-3eca7188="" aria-hidden="true" focusable="false" data-prefix="fas" data-icon="angle-right" role="img" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 256 512" data-fa-i2svg=""><path fill="currentColor" d="M64 448c-8.188 0-16.38-3.125-22.62-9.375c-12.5-12.5-12.5-32.75 0-45.25L178.8 256L41.38 118.6c-12.5-12.5-12.5-32.75 0-45.25s32.75-12.5 45.25 0l160 160c12.5 12.5 12.5 32.75 0 45.25l-160 160C80.38 444.9 72.19 448 64 448z"></path></svg>`

when removing 'fa' after 'svg-inline--', the **transition** works   but my icon is huge in size

 ````
.dropdown{
  transition: 0.3s ease;
 }

 .dropdown.rotate {
     transform: rotate(90deg);
 } 
````
